src/matrix/test_data/generate_numpy_array_test_data.py

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env python3
+
+# Copyright 2020   Mobvoi AI Lab, Beijing, China (author: Fangjun Kuang)
+# Apache 2.0
+
+import numpy as np
+
+
+def main():
+    data = np.arange(8).astype(np.float32)
+    np.save('float_vector.npy', data)
+    np.save('double_vector.npy', data.astype(np.float64))
+
+    data = data.reshape(2, 4).astype(np.float32)
+    np.save('float_matrix.npy', data)
+    np.save('double_matrix.npy', data.astype(np.float64))
```

Comment:
The 'npy' files should be removed in the commit if they are generated during test.

Reply:
My intention was that the test is run by `make test` and the user needn't generate
the test data in advance. In addition, it needs to install `numpy` to run this file
and not everyone using kaldi has `numpy` installed on his/her computer.

Reply:
I would rather check in some. Otherwise, the read test gets the dependency on the write test: you cannot run the read test in isolation, and also if the write test fails. Small binary files are totally ok.

Reply:
Better than a dependency of the test on numpy. We do not have such a dependency now, so pre-generated files are better by me.

Reply:
Pre-generated files are checked in along with this pullrequest.

Reply:
Thanks, it's better to decouple them this way. I discovered I had no numpy once I attempted to generate training progress plots of a model that was unstable, a year and a half of using that machine for training; usually I just gleaned all the progress info from logs (when everything converges a expected, it's easy to grep and see the pattern). So numpy by itself is a very non-requirement for Kaldi. :)